### PR TITLE
docs: add release checklist

### LINKED ISSUE #287: docs: add a release checklist
We should write down the process to cut a new release. @itowlson mentioned the documentation should be updated to refer to the `v` tag prefix for creating a release, for example. That might not be totally clear to other contributors looking to cut their first release.

### PR DESCRIPTION
**Pull request by bacongobbler**
closes #287

### CODE REVIEW

File: docs/community/release-checklist.md [hunk 1]
```diff
@@ -0,0 +1,340 @@
+# Release Checklist
+
+> If your experience deviates from this document, please document the changes to keep it up-to-date.
+
+This guide will guide you (pun intended) on your journey to release a new version of Krustlet.
```
Comment (thomastaylor312) [inline docs/community/release-checklist.md:5]: Suggested edit: "redundancy added for purposes of punnery"

File: docs/community/release-checklist.md [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+
+It is important to note that this document assumes that the git remote in your repository that
+corresponds to <https://github.com/deislabs/krustlet> is named `upstream`. If yours is not (for
+example, if you've chosen to name it `origin` or something similar instead), be sure to adjust the
+listed snippets for your local environment accordingly.
+
+If you are not sure what your upstream remote is named, use a command like `git remote -v` to find
+out.
+
+If you don't have an upstream remote, you can add one easily using something like:
```
Comment (thomastaylor312) [inline docs/community/release-checklist.md:19]: In this case, I don't think it is "something like," I think this should be the command they run if they don't have an `upstream`

File: docs/community/release-checklist.md [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+There are two versions of the checklist: one for patch releases, and one for major/minor releases.
+Follow the checklist that best applies below.
+
+## Checklist for major and minor releases
+
+Follow this checklist if you are cutting a major release (vX.0.0) or a minor release (vX.Y.0):
+
+1. set up your environment
+1. increment the version number
+1. create the release branch
```
Comment (thomastaylor312) [inline docs/community/release-checklist.md:54]: Is this something we want to start doing now? I figured we'd start doing this after we hit 1.0

File: docs/community/release-checklist.md [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+
+Download Krustlet vX.Y.Z:
+
+- [checksums-vX.Y.Z.txt](https://krustlet.blob.core.windows.net/releases/checksums-vX.Y.Z.txt)
+- [krustlet-vX.Y.Z-linux-amd64.tar.gz](https://krustlet.blob.core.windows.net/releases/krustlet-vX.Y.Z-linux-amd64.tar.gz)
+- [krustlet-vX.Y.Z-macos-amd64.tar.gz](https://krustlet.blob.core.windows.net/releases/krustlet-vX.Y.Z-macos-amd64.tar.gz)
+```
+
+Feel free to bring in your own personality into the release notes; it's nice for people to think
+we're not all robots. =3
```
Comment (thomastaylor312) [inline docs/community/release-checklist.md:205]: Is this supposed to be `:)`?

File: docs/community/release-checklist.md (hunk 3 above)
Comment (bacongobbler) [reply docs/community/release-checklist.md:54]: yeah. Most of this is copied over from the Helm release checklist. Many of the items in this list don't apply. We can add these steps after we hit 1.0. I'll remove those steps for now.

File: docs/community/release-checklist.md (hunk 4 above)
Comment (bacongobbler) [reply docs/community/release-checklist.md:205]: https://www.urbandictionary.com/define.php?term=%3A3

I'll change this to a winky face. It's not a well-known symbol.

**Pull request merged**